cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AuthorizationModule.java

```diff
@@ -0,0 +1,160 @@
+import co.cask.cdap.security.spi.authorization.TransactionalFunctionExecutor;
+import co.cask.cdap.security.spi.authorization.TxCallable;
+import co.cask.tephra.TransactionContext;
+import co.cask.tephra.TransactionFailureException;
+import co.cask.tephra.TransactionSystemClient;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Inject;
+import com.google.inject.PrivateModule;
+import com.google.inject.Provider;
+import com.google.inject.Scopes;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+/**
+ * {@link PrivateModule} for authorization classes.
+ */
+public class AuthorizationModule extends PrivateModule {
+  @Override
+  protected void configure() {
+    bind(DatasetContext.class).toProvider(DatasetContextProvider.class).in(Scopes.SINGLETON);
+    expose(DatasetContext.class);
+
+    bind(Admin.class).toProvider(AdminProvider.class);
+    expose(Admin.class);
+
+    bind(TransactionalFunctionExecutor.class).toProvider(TransactionalFunctionExecutorProvider.class);
+    expose(TransactionalFunctionExecutor.class);
+
+    bind(AuthorizerInstantiatorService.class).in(Scopes.SINGLETON);
+    expose(AuthorizerInstantiatorService.class);
+  }
+
+  private static final class DatasetContextProvider implements Provider<DynamicDatasetCache> {
+
+    private final DatasetFramework dsFramework;
+    private final TransactionSystemClient txClient;
+    private final MetricsCollectionService metricsCollectionService;
+
+    @Inject
+    private DatasetContextProvider(DatasetFramework dsFramework, TransactionSystemClient txClient,
+                                   MetricsCollectionService metricsCollectionService) {
+      this.dsFramework = dsFramework;
+      this.txClient = txClient;
+      this.metricsCollectionService = metricsCollectionService;
+    }
+
+    @Override
+    public DynamicDatasetCache get() {
+      SystemDatasetInstantiator dsInstantiator = new SystemDatasetInstantiator(dsFramework, null, null);
+      return new MultiThreadDatasetCache(
+        dsInstantiator, txClient, NamespaceId.SYSTEM, ImmutableMap.<String, String>of(),
+        metricsCollectionService.getContext(ImmutableMap.<String, String>of()),
+        ImmutableMap.<String, Map<String, String>>of()
+      );
+    }
+  }
+
+  private static final class AdminProvider implements Provider<Admin> {
+    private final DatasetFramework dsFramework;
+
+    @Inject
+    private AdminProvider(DatasetFramework dsFramework) {
+      this.dsFramework = dsFramework;
+    }
+
+    @Override
+    public Admin get() {
+      return new DefaultAdmin(dsFramework, NamespaceId.SYSTEM);
+    }
+  }
+
+  private static final class TransactionalFunctionExecutorProvider implements Provider<TransactionalFunctionExecutor> {
+
+    private final Logger log = LoggerFactory.getLogger(TransactionalFunctionExecutorProvider.class);
```

Comment:
Not sure what I was thinking. I probably got an error because I had not made the class static in the beginning so intellij complained about the static modifier for this field. Fixed.